File: programmers_네트워크.py
def solution(n, computers):
    answer = []
    for i in range(n):
        visit = [0]*n
        stack = [i]
        while stack != []:
            x = stack.pop()
            for z in range(n):
                if computers[x][z]==1 and visit[z] == 0:
                    visit[z]=1
                    stack.append(z)
        if visit not in answer:
            answer.append(visit)
    return len(answer)

print(solution(3,[[1,1,0],[1,1,0],[0,0,1]]))
print(solution(3,[[1,1,0],[1,1,1],[0,1,1]]))

File: programmers_네트워크.js
function solution(n, computers) {
  var answer = 0;
  const visitStatus = new Array(n);
  for (var i = 0; i < n; i++) {
    if (visitStatus[i] !== undefined) continue;
    const innerVisit = new Array(n);
    innerVisit[i] = true;
    const stack = [i];
    while (stack.length > 0) {
      let node = stack.pop();
      for (var j = 0; j < n; j++) {
        if (computers[node][j] === 0) continue;
        if (innerVisit[j] !== undefined) continue;
        visitStatus[j] = true;
        innerVisit[j] = true;
        stack.push(j);
      }
    }
    answer = answer + 1;
  }
  return answer;
}
